controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+			return err
+		}
+		r.Log.Info("Creating or updating odf-info configmap", OdfInfoMapKind, klog.KRef(storageCluster.Namespace, OdfInfoConfigMapName))
+		odfInfoData, configErr := getOdfInfoData(r, storageCluster)
+		if configErr != nil {
+			return fmt.Errorf("failed to get ODF info config data: %w", configErr)
+		}
+		odfInfoKey := obj.getOdfInfoKeyName(storageCluster)
+		// Creates or appends to the data map
+		odfInfoConfigMap.Data[odfInfoKey] = odfInfoData
+		return nil
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update odf-info config", OdfInfoMapKind, klog.KRef(storageCluster.Namespace, OdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update odf-info config: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *odfInfoConfig) getOdfInfoKeyName(storageCluster *ocsv1.StorageCluster) string {
+	return storageCluster.Namespace + "/" + storageCluster.Name + "." + OdfInfoKeyName
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	odfInfoConfigMap := &corev1.ConfigMap{}
+	odfInfoConfigMap.ObjectMeta.Name = OdfInfoConfigMapName
+	odfInfoConfigMap.ObjectMeta.Namespace = operatorNamespace
+	if err = r.Client.Get(r.ctx, client.ObjectKeyFromObject(odfInfoConfigMap), odfInfoConfigMap); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	if len(odfInfoConfigMap.Data) > 1 {
+		delete(odfInfoConfigMap.Data, obj.getOdfInfoKeyName(storageCluster))
+		if err = r.Client.Update(r.ctx, odfInfoConfigMap); err != nil {
+			r.Log.Error(err, "Failed to delete key in odf-info.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete key %v in odf-info %v: %v", obj.getOdfInfoKeyName(storageCluster), odfInfoConfigMap.Name, err)
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := OdfDeploymentTypeExternal
+	isDROptimized := false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
```

Comment:
This comment does not match the `if` that comes after it. 
```suggestion
	// if this cluster represents an internal mode deployment
```

Reply:
As I'm not the one who wrote this function, I think @umangachapagain can provide input regarding the accuracy of your suggestion.